### ISSUE TITLE
[harbor_rl] Binary test fixtures silently zero out 7 Terminal-Bench tasks

### ISSUE THREAD

**Issue by miguelrc-scale** [thinking-machines-lab/tinker-cookbook#889]
### Summary

`HarborReward._upload_tests` reads every file under a task's `tests/` directory as text:

https://github.com/thinking-machines-lab/tinker-cookbook/blob/main/tinker_cookbook/recipes/harbor_rl/harbor_tools.py#L100

```python
content = file_path.read_text()
```

Terminal-Bench tasks whose graders compare against binary ground truth raise `UnicodeDecodeError` here. `HarborReward.__call__` catches it and returns `reward=0.0, grading_error=1.0`, so those tasks score zero no matter what the agent did.

### Affected tasks

7 of the 89 tasks in `terminal-bench@2.0`:

| Task | Binary fixture |
|---|---|
| `build-pov-ray` | `tests/reference_illum1.png` |
| `make-doom-for-mips` | `tests/reference.jpg` |
| `make-mips-interpreter` | `tests/reference.jpg` |
| `pytorch-model-recovery` | `tests/weights_gtruth.pt` |
| `sam-cell-seg` | `tests/test_img.png` |
| `train-fasttext` | `tests/private_test.tar.gz` |
| `video-processing` | `tests/test_video.mp4` |

These fixtures are load-bearing, not incidental — `sam-cell-seg/tests/test.sh:25` does `cp /tests/test_img.png .`, and `make-mips-interpreter/tests/test_outputs.py:71` reads `/tests/reference.jpg` to run an L2 image comparison against the agent's output.

### Observed

From a `train_terminal_bench.py` run on `moonshotai/Kimi-K2.6`:

```
harbor_tools:87 [ERROR] Harbor grading failed: 'utf-8' codec can't decode byte 0x89 in position 0: invalid start byte
harbor_tools:87 [ERROR] Harbor grading failed: 'utf-8' codec can't decode byte 0xff in position 0: invalid start byte
harbor_tools:87 [ERROR] Harbor grading failed: 'utf-8' codec can't decode byte 0x8b in position 1: invalid start byte
```

`0x89` = PNG, `0xff` = JPEG, `0x8b` = gzip — matching the table above.

### Why this is easy to miss

The grading failure is swallowed rather than raised, so it never surfaces as an error. In `eval.py`:

```python
status = "ERROR" if result.error else ("PASS" if result.reward > 0 else "FAIL")
```

A returned `0.0` is classified `FAIL`, indistinguishable from a genuine miss. In the README's Terminal-Bench 2.0 results (14 PASS / 31 FAIL / 44 ERROR) these would sit inside the 31 FAILs.

For RL specifically it is worse than a lost eval sample: because the reward is constant zero across the group, advantages center to zero and the group contributes no gradient, so the sandboxes are paid for and produce nothing.

### Fix

Read as bytes. `SandboxInterface.write_file` already accepts `str | bytes`, and `ModalSandbox.write_file` encodes `str` to bytes before streaming to stdin, so bytes is the path it was already built for.

PR with a regression test to follow.

### Environment

- `tinker-cookbook` @ `main` (also present on `nightly`; the two are identical for this file)
- dataset: `harbor datasets download terminal-bench@2.0`

**Comment by IgorGanapolsky** [comment]
## Repro note

Leaderboard and cookbook scores only stay comparable when **protocol is pinned** (system prompt, max_tokens, temperature, timeout, renderer, thinking-token handling). Tinker’s eval README already flags this sensitivity — linking each public result to the exact `BenchmarkConfig` + renderer avoids “verification” against a different protocol.

For agent post-training we also separate:

- **LLM-as-a-Judge / rubrics** for open-ended quality
- **deterministic claim validators** for cash/LIVE/tool success (fail-closed)

Happy to help cross-check a specific row against documented cookbook settings.

_Issue #889: [harbor_rl] Binary test fixtures silently zero out 7 Terminal-Bench tasks_

---
*Independent of Thinking Machines Lab. Related open work: [ThumbGate.app](https://thumbgate.app/) (Hermes agent control plane) and fail-closed GTM/cash validators in [mac-yolo-safeguards](https://github.com/IgorGanapolsky/mac-yolo-safeguards) (`tools/tinker-brain/`, `tools/governed-data-mesh.js`). No affiliation claimed with TML or model providers.*
